### PR TITLE
Handle XAS response and tests

### PR DESCRIPTION
**Pull request by sbenedicadb**

### CODE REVIEW

File: AEPMessaging/Sources/PropositionPayload.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    
+    init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        let id = try values.decode(String.self, forKey: .id)
+        let scope = try values.decode(String.self, forKey: .scope)
+        let scopeDetails = try values.decode([String: AnyCodable].self, forKey: .scopeDetails)
+        
+        propositionInfo = PropositionInfo(id: id, scope: scope, scopeDetails: scopeDetails)
+        items = try values.decode([PayloadItem].self, forKey: .items)
```
Comment (swarna04) [inline AEPMessaging/Sources/PropositionPayload.swift:34]: caution: An issue with array decode is that it's all or none. Would you have a scenario where you need to ignore one or more (invalid) payload items in the array?

File: AEPMessaging/Sources/PayloadItem.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+
+struct PayloadItem: Codable {
+    var id: String?
+    var schema: String?
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply AEPMessaging/Sources/PayloadItem.swift:17]: messaging extension doesn't use them

File: AEPMessaging/Sources/ItemData.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+
+struct ItemData: Codable {
+    var id: String?
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply AEPMessaging/Sources/ItemData.swift:16]: messaging extension doesn't use it

File: AEPMessaging/Sources/PropositionPayload.swift (hunk 1 above)
Comment (sbenedicadb) [reply AEPMessaging/Sources/PropositionPayload.swift:34]: it's a good callout - to be honest if we end up in a position where only some of the items are invalid, i'd have greater concerns about something being broken in the process.  in that case, i think it's best to just not load any messages. wdyt?

**Review by swarna04 — APPROVED** [review]

**Pull request merged**